gadgets/grappa/GrappaGadget.cpp

```diff
@@ -346,14 +346,20 @@ namespace Gadgetron{
       image_data_[slice] = 0;
       if (create_image_buffer(slice) != GADGET_OK) {
         GDEBUG("Failed to create image buffer");
+		cm0->release();
+		cm1->release();
         return GADGET_FAIL;
       }
 
       if (this->next()->putq(cm0) < 0) {
         GDEBUG("Failed to pass image on to next Gadget in chain\n");
+		cm0->release();
+		cm1->release();
         return GADGET_FAIL;
       }
-
+	  cm0->release();
```

Comment:
Just as an example, you should not release cm0 if you have succeeded in passing it on to the next gadget.